graphistry/pygraphistry.py

```diff
@@ -2446,7 +2461,140 @@ def _handle_api_response(response):
+        """
+        
+        for _ in range(repeat):
+            token = PyGraphistry.sso_get_token()
+            if token:
+                return token
+            time.sleep(wait)
+
+        return
+
+    @staticmethod
+    def sso_wait_for_token_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False, display_mode: str = 'text'):
+        if display_mode == 'html':
+            PyGraphistry.sso_wait_for_token_html_display(repeat, wait, fail_silent)
+        else:
+            PyGraphistry.sso_wait_for_token_text_display(repeat, wait, fail_silent)
+
+    @staticmethod
+    def sso_wait_for_token_text_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False):
+        """Get the JWT token for SSO login and display corresponding message in text
+        Get the JWT token for SSO login and display corresponding message in text
+        """
+        if not PyGraphistry.api_token():
+            msg_text = '....'
+            if not PyGraphistry.sso_repeat_get_token(repeat, wait):
+                msg_text = f'{msg_text}\nFailed to get token after {repeat*wait} seconds ....'
+                if not fail_silent:
+                    msg = f"Failed to get token after {repeat*wait} seconds. Please re-run the login process"
+                    if in_ipython() or in_databricks() or PyGraphistry.set_sso_opt_into_type == "display":
+                        display_message_html(f"<strong>{msg}</strong>")
+                    raise Exception(msg) 
+                else:
+                    msg_text = f'{msg_text}\nGot token'
+                    print(msg_text)
+                    return
+
+            msg_text = f'{msg_text}\nGot token'
+            print(msg_text)
+        else:
+            print('Token is valid, no waiting required.')
+
+
+    @staticmethod
+    def sso_wait_for_token_html_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False):
+        """Get the JWT token for SSO login and display corresponding message in HTML
+        Get the JWT token for SSO login and display corresponding message in HTML
+        """
+        from IPython.display import display, HTML
+        if not PyGraphistry.api_token():
+            msg_html = '<br /><strong> .... </strong>'
+            if not PyGraphistry.sso_repeat_get_token(repeat, wait):
+                msg_html = f'{msg_html}<br /><strong>Failed to get token after {repeat*wait} seconds .... </strong>'
+                if not fail_silent:
+                    raise Exception(f"Failed to get token after {repeat*wait} seconds. Please re-run the login process") 
+                else:
+                    msg_html = f'{msg_html}<br /><strong>Got token</strong>'
+                    display(HTML(msg_html))
+                    return
+
+            msg_html = f'{msg_html}<br /><strong>Got token</strong>'
+            display(HTML(msg_html))
+        else:
+            display(HTML('<br /><strong>Token is valid, no waiting required.</strong>'))
```

Comment:
`display(HTML('<br /><strong>Token is valid; no further action needed.</strong>'))
`